playbook/app/pb_kits/playbook/pb_multiple_users_stacked/_multiple_users_stacked.scss

```diff
@@ -3,6 +3,320 @@
 @import "../tokens/opacity";
 @import "../pb_avatar/avatar";
 
+$avatar-sizes: (
```

Comment:
@markdoeswork I did a little refactoring on your .scss and wanted to see what you thought! 
- I was able to save 338 lines on your `.scss`, by leveraging some functions from SCSS
- I was also able to use our spacing tokens on tons of spots so that we don't have to use hardcoded values
- I fully tested the code and compared it to your visual on the docs and it looked good

You can commit directly from here in github if you want...
Here's the code:

```scss
@import "../tokens/colors";
@import "../tokens/border_radius";
@import "../tokens/opacity";
@import "../tokens/spacing";
@import "../pb_avatar/avatar";

// Define maps for sizes and positions
$sizes: (
  "avatar": (sm: $space_xl - $space_xxs, md: $space_xl + $space_md, lg: $space_xl * 2, xl: $space_xl * 2 + $space_md),
  "first-item": (
    "double": (sm: $space_md - $space_xxs, md: $space_lg, lg: $space_xl + $space_xxs, xl: $space_xl + $space_sm),
    "triple": (sm: $space_sm, md: $space_md, lg: $space_lg, xl: $space_xl + $space_xxs),
    "quadruple": (sm: $space_sm, md: $space_lg - $space_xxs, lg: $space_lg + $space_xxs, xl: $space_xl + $space_xxs)
  ),
  "second-item": (
    "double": (sm: $space_sm - $space_xxs, md: $space_sm, lg: $space_md - $space_xxs, xl: $space_md),
    "triple": (sm: $space_sm - $space_xxs, md: $space_md - $space_xxs, lg: $space_md, xl: $space_lg),
    "quadruple": (sm: $space_sm - $space_xxs, md: $space_md - $space_xxs, lg: $space_lg - $space_xxs, xl: $space_lg)
  ),
  "third-item": (
    "triple": (sm: 10px, md: $space_sm, lg: $space_md - $space_xxs, xl: $space_md),
    "quadruple": (sm: 10px, md: $space_sm, lg: $space_md, xl: $space_md)
  ),
  "fourth-item": (
    "quadruple": (sm: $space_xxs, md: $space_sm - $space_xxs, lg: $space_sm, xl: $space_sm)
  )
);

$positions: (
  "first-item": (
    "double": (sm: (top: $space_xxs, left: $space_xxs), md: (top: $space_xs, left: $space_xs), lg: (top: $space_xs, left: $space_xs), xl: (top: 10px, left: 10px)),
    "triple": (sm: (top: $space_xxs, left: $space_xxs), md: (top: $space_xs, left: $space_xs), lg: (top: 10px, left: 10px), xl: (top: $space_sm - $space_xxs, left: $space_sm - $space_xxs)),
    "quadruple": (sm: (top: $space_xs, left: $space_xxs), md: (top: $space_xs, left: $space_xs), lg: (top: $space_sm, left: $space_xs), xl: (top: $space_sm, left: 10px))
  ),
  "second-item": (
    "double": (sm: (bottom: $space_xs, right: $space_xxs), md: (bottom: 10px, right: $space_xs), lg: (top: $space_xl + $space_sm - $space_xxs, right: $space_sm - $space_xxs), xl: (top: $space_xl + $space_sm + $space_xs, right: $space_sm - $space_xxs)),
    "triple": (sm: (top: $space_sm - $space_xxs, right: $space_xxs), md: (top: $space_md, right: $space_xs), lg: (top: $space_lg, right: $space_sm), xl: (top: $space_xl + $space_xs, right: 11px)),
    "quadruple": (sm: (bottom: $space_sm, right: $space_xxs), md: (top: $space_md, right: $space_xs), lg: (top: $space_lg - $space_xs, right: $space_xs), xl: (top: $space_xl + $space_xxs, right: $space_sm))
  ),
  "third-item": (
    "triple": (sm: (bottom: $space_xxs, left: 11px), md: (bottom: $space_xs, left: $space_sm), lg: (bottom: 10px, left: $space_md - $space_xs), xl: (bottom: $space_sm - $space_xxs, left: $space_lg - $space_xs)),
    "quadruple": (sm: (bottom: $space_xxs, left: $space_sm), md: (bottom: $space_xs, left: $space_sm), lg: (bottom: $space_xs, left: $space_md - $space_xxs), xl: (bottom: 11px, left: $space_lg - $space_xs))
  ),
  "fourth-item": (
    "quadruple": (sm: (top: $space_xs, right: $space_xs), md: (top: $space_xs, right: $space_sm - $space_xxs), lg: (top: $space_sm, right: $space_sm), xl: (top: $space_sm, right: $space_md))
  )
);

// Functions and Mixins
@function get-size($item, $type, $size) {
  @return map-get(map-get(map-get($sizes, $item), $type), $size);
}

@function get-position($item, $type, $size) {
  @return map-get(map-get(map-get($positions, $item), $type), $size);
}

@mixin avatar-size($size) {
  height: $size;
  width: $size;
  .avatar_wrapper { width: $size; height: $size; }
}

@mixin position($position) {
  position: absolute;
  @each $pos, $val in $position { #{$pos}: $val; }
}

@mixin avatar-styles($size, $border-color: $white) {
  @include avatar-size($size);
  .avatar_wrapper { border: 1px solid $border-color; }
}

@mixin generate-bubble-styles($size-name, $size-value) {
  @include avatar-size($size-value);
  width: $size-value;
  height: $size-value;
  flex-basis: $size-value;
  background-color: $bg_light;
  border-radius: 50%;

  &.dark { background-color: $card_dark; }

  [class^=pb_avatar_kit].pb_multiple_users_stacked_item {
    @include avatar-styles($size-value * 0.45);
    &.dark .avatar_wrapper { border-color: $border_dark; }
  }

  @each $item in ("first", "second", "third", "fourth") {
    [class^=pb_avatar_kit].#{$item}_item {
      @each $bubble_type in ("double", "triple", "quadruple") {
        &.#{$bubble_type}_bubble {
          @include position(get-position("#{$item}-item", $bubble_type, $size-name));
          @include avatar-size(get-size("#{$item}-item", $bubble_type, $size-name));
        }
      }
    }
  }

  &[class*=_single_bubble] {
    [class^=pb_avatar_kit].first_item {
      @include position((top: 0, left: 0));
      @include avatar-size($size-value);
    }
  }
}

// Main styles
[class^=pb_multiple_users_stacked_kit] {
  $container_size: map-get(map-get($sizes, "avatar"), "sm");
  $stacked_size: $space_lg - $space_xxs;

  display: inline-flex;
  width: $space_lg + $space_xxs;
  height: $space_lg + $space_xxs;
  flex-basis: $container_size;
  position: relative;
  flex-shrink: 0;
  flex-grow: 0;

  [class^=pb_avatar_kit].pb_multiple_users_stacked_item {
    @include avatar-styles($stacked_size);
    &.dark .avatar_wrapper { border-color: $bg_dark; }
    img { z-index: 0; }
  }

  &[class*=_single] .pb_multiple_users_stacked_item {
    @include avatar-size($space_lg + $space_xxs);
  }

  [class^=pb_avatar_kit].second_item, [class^=pb_badge_kit].second_item {
    @include position((bottom: 0, right: 0));
    z-index: 2;
    background: tint($primary, 90%);
    border-radius: $border_rad_mega;
    padding: 0;
    span { transform: translateY(0); }
    @include avatar-styles($stacked_size);
  }

  .stacked_item .avatar_wrapper::before {
    font-size: 0;
    color: transparent;
  }

  // Bubble styles
  @each $size_name, $size_value in map-get($sizes, "avatar") {
    &[class*=_bubble_][class*=_size_#{$size_name}] {
      @include generate-bubble-styles($size-name, $size-value);
    }
  }
}
```

Reply:
I think its better to leave these as numbers instead of tokens because is more exact to the handoff https://www.figma.com/design/Q1mz4nRApKN5rlzqCV4IC0/Multiple-User-Bubble?node-id=3701-1514&node-type=frame&t=UzPFUVdKYFXlbaEQ-0

$space_xl + $space_md = 64 but we want this to actually be 60 in some instances 

also `$space_xl + $space_sm + $space_xs` is equal to 64 but we really want the top to be 58

@jasperfurniss

Reply:
Using numbers is explicit and better for this case because our designer hand drew these bubbles. 

There isn't a formula where avatar 1 is always 5% from the top. All of these scenarios are explicitly crafted to be their exact size.